qiskit_machine_learning/algorithms/effective_dimension.py

```diff
@@ -0,0 +1,371 @@
+            (
+                self._num_inputs * self._num_params,
+                self._model.output_shape[0],
+                self._model.num_weights,
+            )
+        )
+        outputs = np.zeros((self._num_inputs * self._num_params, self._model.output_shape[0]))
+
+        for (i, param_set) in enumerate(self.params):
+            t_before_forward = time.time()
+            fwd_pass = np.asarray(self._model.forward(input_data=self.inputs, weights=param_set))
+            t_after_forward = time.time()
+
+            if self._callback is not None:
+                msg = f"iteration {i}, time forward pass: {t_after_forward - t_before_forward}"
+                self._callback(msg)
+
+            back_pass = np.asarray(
+                self._model.backward(input_data=self.inputs, weights=param_set)[1]
+            )
+            t_after_backward = time.time()
+
+            if self._callback is not None:
+                msg = f"iteration {i}, time backward pass: {t_after_backward - t_after_forward}"
+                self._callback(msg)
+
+            grads[self._num_inputs * i : self._num_inputs * (i + 1)] = back_pass
+            outputs[self._num_inputs * i : self._num_inputs * (i + 1)] = fwd_pass
+
+        # post-processing in the case of OpflowQNN output, to match the CircuitQNN output format
+        if isinstance(self._model, OpflowQNN):
+            grads = np.concatenate([grads / 2, -1 * grads / 2], 1)
+            outputs = np.concatenate([(outputs + 1) / 2, (1 - outputs) / 2], 1)
+
+        return grads, outputs
+
+    def get_fisher(self, gradients: np.ndarray, model_outputs: np.ndarray) -> np.ndarray:
+
+        """
+        This method computes the average Jacobian for every set of gradients and
+        model output as shown in Abbas et al.
+
+        Args:
+            gradients: A numpy array, result of the neural network's backward pass, of
+                shape ``(num_inputs * num_params, output_size, num_weights)``
+            model_outputs: A numpy array, result of the neural networks' forward pass,
+                of shape ``(num_inputs * num_params, output_size)``
+        Returns:
+            normalized_fisher: A numpy array of shape
+                ``(num_inputs * num_params, num_weights, num_weights)``
+                with the average Jacobian  for every set of gradients and model output given.
+        """
+
+        if model_outputs.shape < gradients.shape:
+            # add dimension to model outputs for broadcasting
+            model_outputs = np.expand_dims(model_outputs, axis=2)
+
+        # get grad-vectors (gradient_k/model_output_k)
+        # multiply by sqrt(model_output) so that the outer product cross term is correct
+        # after Einstein summation
+        gradvectors = np.sqrt(model_outputs) * gradients / model_outputs
+
+        # compute sum of matrices obtained from outer product of grad-vectors
+        fisher = np.einsum("ijk,lji->ikl", gradvectors, gradvectors.T)
```

Comment:
```suggestion
        fisher_information = np.einsum("ijk,lji->ikl", gradvectors, gradvectors.T)
```
Please check other places.